framework/db/ActiveRecordInterface.php

```diff
@@ -427,4 +427,19 @@ public function unlink($name, $model, $delete = false);
      * @return mixed the database connection used by this AR class.
      */
     public static function getDb();
+
+    /**
+     * Creates an active record instance.
+     *
+     * This method is called internally by framework for instantiating new record object.
+     * It is not meant to be used for creating new records directly.
+     *
+     * You may override this method if the instance being created
+     * depends on the row data to be populated into the record.
+     * For example, by creating a record based on the value of a column,
+     * you may implement the so-called single-table inheritance mapping.
+     * @param array $row row data to be populated into the record (can be empty).
+     * @return ActiveRecordInterface the newly created active record
+     */
+    public static function instantiate($row = []);
```

Comment:
Hello @samdark ,

I think having `instantiate` method as is and add a new method that should simply create an empty model for metadata is the best approach right now. 
Right now people use "instantiate" methods to instantiate models and don't expect changes in the method signature or it's meaning.

But these two methods have different meanings and, from my point of view, should not share **"instantiate"** word in names. 
_So I suggest having a method that creates empty models with a name such as:_
- createEmpty
- createNew
- createPlainModel
- create
- newEmptyModel
- newPlainModel

People probably would want to use this method passing instantiation stuff so I don't think **"internal"** or something similar should be used in this method. 

Also, if skip options with **"Model"**, I think, it makes sense to move method to some of the base classes.
For example, `createNew` might be implemented in Object.
`Object::createNew` is pretty natural from my point of view.

So what do you think?